Remove WebArchiveDebugMode

sysrqb opened the pull request:
#### 1eb47513afb8f84f48d36c0daae0eff1bc9b9d03
<pre>
Remove WebArchiveDebugMode
<a href="https://bugs.webkit.org/show_bug.cgi?id=245488">https://bugs.webkit.org/show_bug.cgi?id=245488</a>
&lt;rdar://problem/88406961&gt;

Reviewed by Geoffrey Garen.
Original patch by: Kate Cheney &lt;katherine_cheney@apple.com&gt;

Remove use case of WebArchiveDebugMode and deprecate it now that it
is no longer in use. Add a new temporary flag WebArchiveTestMode
that is more aptly named and will maintain existing behavior for
internal tests temporarily.

In addition, for clarity annotate resource URLs in WebArchive with
&quot;webarchive+&quot; prefix, but avoid breaking web content that transits the
pasteboard. If a requested resource has that scheme prefix, then strip it.

* LayoutTests/platform/wk2/webarchive/loading/test-loading-archive-subresource-null-mimetype-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-subresource-null-mimetype-expected.txt:
* Source/WTF/Scripts/Preferences/WebPreferences.yaml:
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::extractMarkupAndArchive):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::isLoadingRemoteArchive const):
(WebCore::DocumentLoader::commitData):
(WebCore::DocumentLoader::scheduleArchiveLoad):
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/ResourceLoader.cpp:
(WebCore::ResourceLoader::start):
* Source/WebCore/loader/archive/ArchiveResourceCollection.cpp:
(WebCore::ArchiveResourceCollection::addResource):
(WebCore::ArchiveResourceCollection::archiveResourceForURL):
* Source/WebCore/loader/archive/ArchiveResourceCollection.h:
* Source/WebCore/loader/archive/cf/LegacyWebArchive.cpp:
(WebCore::LegacyWebArchive::createResource):
(WebCore::LegacyWebArchive::create):
(WebCore::LegacyWebArchive::extract):
* Source/WebCore/loader/archive/cf/LegacyWebArchive.h:
* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::shouldTreatAsOpaqueOrigin):
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesSetWebArchiveTestingModeEnabled):
(WKPreferencesGetWebArchiveTestingModeEnabled):
(WKPreferencesSetWebArchiveDebugModeEnabled):
(WKPreferencesGetWebArchiveDebugModeEnabled):
* Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferences.mm:
(-[WKPreferences _setWebArchiveTestingModeEnabled:]):
(-[WKPreferences _webArchiveTestingModeEnabled]):
(-[WKPreferences _setWebArchiveDebugModeEnabled:]):
(-[WKPreferences _webArchiveDebugModeEnabled]):
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(+[WebView _canHandleRequest:forMainFrame:]):
* Tools/TestWebKitAPI/Tests/mac/LoadWebArchive.mm:
(TestWebKitAPI::constructArchive):
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/255705@main">https://commits.webkit.org/255705@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ce32dd9dd5e75c254e938e9d0ec76bf4d6394feb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/91489 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/605 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22135 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101220 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/161246 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/95494 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/635 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29384 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83833 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97572 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97147 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/408 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78205 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/27366 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/82332 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81991 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/70404 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/82930 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35619 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16020 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/77974 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33404 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17113 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26928 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3947 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37208 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39903 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/80578 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39125 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36230 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17659 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

geoffreygaren reviewed (approved):
r=me

geoffreygaren commented on Tools/TestWebKitAPI/Tests/mac/LoadWebArchive.mm:
```diff
@@ -241,7 +242,7 @@ - (WKWebView *)webView:(WKWebView *)webView createWebViewWithConfiguration:(WKWe
 TEST(LoadWebArchive, DisallowedNetworkHosts)
 {
-    NSData *data = constructArchive();
+    NSData *data = constructArchive("alert('loaded http subresource successfully')");
```
It would be nice to add a test that verifies that a network request is treated as cross-origin from the perspective of CORS headers, and can be rejected by the server on those grounds.

sysrqb committed:
Remove WebArchiveDebugMode
https://bugs.webkit.org/show_bug.cgi?id=245488
<rdar://problem/88406961>

Reviewed by Geoffrey Garen.
Original patch by: Kate Cheney <katherine_cheney@apple.com>

Remove use case of WebArchiveDebugMode and deprecate it now that it
is no longer in use. Add a new temporary flag WebArchiveTestMode
that is more aptly named and will maintain existing behavior for
internal tests temporarily.

In addition, for clarity annotate resource URLs in WebArchive with
"webarchive+" prefix, but avoid breaking web content that transits the
pasteboard. If a requested resource has that scheme prefix, then strip it.

* LayoutTests/platform/wk2/webarchive/loading/test-loading-archive-subresource-null-mimetype-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-subresource-null-mimetype-expected.txt:
* Source/WTF/Scripts/Preferences/WebPreferences.yaml:
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::extractMarkupAndArchive):
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::isLoadingRemoteArchive const):
(WebCore::DocumentLoader::commitData):
(WebCore::DocumentLoader::scheduleArchiveLoad):
* Source/WebCore/loader/DocumentLoader.h:
* Source/WebCore/loader/ResourceLoader.cpp:
(WebCore::ResourceLoader::start):
* Source/WebCore/loader/archive/ArchiveResourceCollection.cpp:
(WebCore::ArchiveResourceCollection::addResource):
(WebCore::ArchiveResourceCollection::archiveResourceForURL):
* Source/WebCore/loader/archive/ArchiveResourceCollection.h:
* Source/WebCore/loader/archive/cf/LegacyWebArchive.cpp:
(WebCore::LegacyWebArchive::createResource):
(WebCore::LegacyWebArchive::create):
(WebCore::LegacyWebArchive::extract):
* Source/WebCore/loader/archive/cf/LegacyWebArchive.h:
* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::shouldTreatAsOpaqueOrigin):
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesSetWebArchiveTestingModeEnabled):
(WKPreferencesGetWebArchiveTestingModeEnabled):
(WKPreferencesSetWebArchiveDebugModeEnabled):
(WKPreferencesGetWebArchiveDebugModeEnabled):
* Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKPreferences.mm:
(-[WKPreferences _setWebArchiveTestingModeEnabled:]):
(-[WKPreferences _webArchiveTestingModeEnabled]):
(-[WKPreferences _setWebArchiveDebugModeEnabled:]):
(-[WKPreferences _webArchiveDebugModeEnabled]):
* Source/WebKit/UIProcess/API/Cocoa/WKPreferencesPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(+[WebView _canHandleRequest:forMainFrame:]):
* Tools/TestWebKitAPI/Tests/mac/LoadWebArchive.mm:
(TestWebKitAPI::constructArchive):
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/255705@main
